### PR TITLE
[MWPW-154335] [callout] Spacing issue encountered when the call-out section is added

### PR DESCRIPTION
**Pull request by rohitsahu**
- Adjust heigh of body if a card has callout text
- Callout text will now be inside one single div. It has been done to keep the height calculation simpler and for consistency with other blocks
- renamed 'callout-text' slot to 'callout-content'
- css fixes
- UTs fixes for above changes
- If there is a pricining inside the text, unset the pricing min height and use the default callout text height

Fix #[MWPW-154335](https://jira.corp.adobe.com/browse/MWPW-154335)

Milo Test link:
https://mwpw-154335--milo--rohitsahu.hlx.page/drafts/rosahu/rosahu

URL for testing:
- https://mwpw-154335--mas--adobecom.hlx.page/

Test URLs:
Before: https://main--milo--adobecom.hlx.page/drafts/rosahu/rosahu
After: https://mwpw-154335--milo--rohitsahu.hlx.page/drafts/rosahu/rosahu

### CODE REVIEW

File: libs/commerce.js [hunk 1]
```diff
@@ -1,4 +1,4 @@
...ing"||typeof f=="number"?String(f):""),a.push({type:typeof f=="string"?U.literal:U.object,value:f});continue}if(et(u)){var d=typeof u.style=="string"?n.date[u.style]:Me(u.style)?u.style.parsedOptions:void 0;a.push({type:U.literal,value:r.getDateTimeFormat(t,d).format(f)});continue}if(tt(u)){var d=typeof u.style=="string"?n.time[u.style]:Me(u.style)?u.style.parsedOptions:void 0;a.push({type:U.literal,value:r.getDateTimeFormat(t,d).format(f)});continue}if(Ke(u)){var d=typeof u.style=="string"?n.number[u.style]:ot(u.style)?u.style.parsedOptions:void 0;d&&d.scale&&(f=f*(d.scale||1)),a.push({type:U.literal,value:r.getNumberFormat(t,d).format(f)});continue}if(it(u)){var h=u.children,E=u.value,S=i[E];if(!so(S))throw new bn(E,"function",s);var N=Fe(h,t,r,n,i,o),P=S(N.map(function(y){return y.value}));Array.isArray(P)||(P=[P]),a.push.apply(a,P.map(function(y){return{type:typeof y=="string"?U.literal:U.object,value:y}}))}if(rt(u)){var b=u.options[f]||u.options.other;if(!b)throw new ir(u.value,f,Object.keys(u.options),s);a.push.apply(a,Fe(b.value,t,r,n,i));continue}if(nt(u)){var b=u.options["="+f];if(!b){if(!Intl.PluralRules)throw new Ge(`Intl.PluralRules is not available in this environment.
```
Comment (rohitsahu) [inline libs/commerce.js:1]: remove this file change from PR, if not needed

File: libs/mas.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
...nstance=Object.defineProperties(s,Object.getOwnPropertyDescriptors({...ri(a,t),...ni(a),...oi(a),...ai(a),...br,Log:M,get defaults(){return T},get literals(){return n},get log(){return M},get providers(){return{checkout(c){return o.checkout.add(c),()=>o.checkout.delete(c)},price(c){return o.price.add(c),()=>o.price.delete(c)}}},get settings(){return i}})),r.debug("Activated:",{literals:n,settings:i,element:s}),document.head.append(s),le(()=>{let c=new CustomEvent(ge,{bubbles:!0,cancelable:!1,detail:k.instance});k.instance.dispatchEvent(c)}),k.instance}function ci(){document.head.querySelector(ce)?.remove(),k.promise=null,M.reset()}function Ur(e,t){if(ae(e)){let r=ae(t)?t():{};return r.force&&ci(),k.promise??(k.promise=ps(e(),r))}return k.promise?k.promise:new Promise(r=>{let n=i=>{r(i.detail)};document.head.addEventListener(ge,n,{once:!0})})}var{origin:ms,searchParams:St}=new URL(import.meta.url),hs=St.get("locale")??"US_en",Dl=St.get("lang")??"en",li=St.get("env")==="stage",ds=St.get("features"),Es=li?"stage":"prod",gs=li?"STAGE":"PROD",xs=()=>({env:{name:Es},commerce:{"commerce.env":gs},locale:{prefix:hs}});Ur(xs);ds.includes("merch-card")&&import(`${ms}/libs/merch-card-all.js`);
```
Comment (rohitsahu) [inline libs/mas.js:3]: remove this file change from PR, if not needed

File: libs/merch-card-all.js [hunk 3]
```diff
@@ -1616,7 +1627,7 @@ body.merch-modal {
...
...is.storageOptions?.selected;if(e){let t=this.querySelector(`merch-offer-select[storage="${e}"]`);if(t)return t}return this.querySelector("merch-offer-select")}get offerSelect(){return this.storageOptions?this.storageSpecificOfferSelect:this.querySelector("merch-offer-select")}get quantitySelect(){return this.querySelector("merch-quantity-select")}merchCardReady(){this.offerSelect&&!this.offerSelect.planType||this.dispatchEvent(new CustomEvent(it,{bubbles:!0}))}handleStorageChange(){let e=this.closest("merch-card")?.offerSelect.cloneNode(!0);e&&this.dispatchEvent(new CustomEvent(we,{detail:{offerSelect:e},bubbles:!0}))}get dynamicPrice(){return this.querySelector('[slot="price"]')}selectMerchOffer(e){if(e===this.merchOffer)return;this.merchOffer=e;let t=this.dynamicPrice;if(e.price&&t){let r=e.price.cloneNode(!0);t.onceSettled?t.onceSettled().then(()=>{t.replaceWith(r)}):t.replaceWith(r)}}};customElements.define(wt,_e);var Z=class extends b{static properties={size:{type:String,attribute:!0},src:{type:String,attribute:!0},alt:{type:String,attribute:!0},href:{type:String,attribute:!0}};constructor(){super(),this.size="m",this.alt=""}render(){let{href:e}=this;return e?l`<a href="${e}">
```
Comment (rohitsahu) [inline libs/merch-card-all.js:1619]: validate this diff

File: web-components/src/global.css.js [hunk 4]
```diff
@@ -251,43 +252,50 @@ merch-card [slot='heading-xl'] {
...
     border-radius: var(--consonant-merch-spacing-xxxs);
     padding: var(--consonant-merch-spacing-xxxs) var(--consonant-merch-spacing-xxxs) var(--consonant-merch-spacing-xxxs) var(--consonant-merch-spacing-xxs);
 }
 
-merch-card [slot='callout-text'] > div.callout-content-wrapper-with-icon {
-    padding-right: var(--consonant-merch-card-callout-icon-padding);
-}
-
-merch-card [slot='callout-text'] > div > div {
+merch-card [slot='callout-content'] > div > div > div {
```
Comment (npeltier) [inline web-components/src/global.css.js:277]: shouldn't we add a class to that last div and avoid rather brittle div > div > div rule?

Comment (rohitsahu) [reply web-components/src/global.css.js:277]: Agreed. Should I also then assign a class to the 2nd last div. I am thinking of something like this:
```
<div slot="callout-content">
 <div>
  <div class="callout-wrapper">
    <div class="callout-text">
    <img>
```

**Comment by rohitsahu** [comment]
@npeltier, few files inside libs, mainly commerce.js and mas.js got updated after running "npm run build". Are these the expected changes?

File: web-components/src/global.css.js [hunk 5]
```diff
@@ -251,43 +252,50 @@ merch-card [slot='heading-xl'] {
     color: var(--merch-color-grey-80);
 }
 
-merch-card [slot='callout-text'] {
-    display: inline-block;
+merch-card [slot='callout-content'] {
+    display: flex;
```
Comment (npeltier) [inline web-components/src/global.css.js:256]: what's the rationale behind all those display: flex just for the sake of adding spaces?

File: web-components/src/global.css.js [hunk 6]
```diff
@@ -251,43 +252,50 @@ merch-card [slot='heading-xl'] {
...
+    flex-direction: column;
     margin: var(--consonant-merch-spacing-xxxs) 0px;
+    gap: var(--consonant-merch-card-callout-spacing-xxs);
 }
 
-merch-card [slot='callout-text'] > div {
-    position: relative;
-    display: inline-grid;
+merch-card [slot='callout-content'] > div {
+    display: flex;
```
Comment (npeltier) [inline web-components/src/global.css.js:263]: here too

File: web-components/src/global.css.js (hunk 5 above)
Comment (rohitsahu) [reply web-components/src/global.css.js:256]: We will require one slot div for entire callout content. 
Then a div to separate the individual callouts. We don't want the background across entire callouts. This div will have texts, links, inline pricing etc inside another div and optionally an image tag.
I had to wrap the above div inside an additional div to place the image on top right. Note that for single line callout text, the image comes at last inline, but for multiline I had to use flex to push the image on right, keeping the text div as inline block.

File: web-components/src/global.css.js (hunk 6 above)
Comment (rohitsahu) [reply web-components/src/global.css.js:263]: This is to prevent the background grey from spanning across entire width of the div

**Review by npeltier — APPROVED** [review]

**Review by Axelcureno — APPROVED** [review]
lgtm

**Pull request merged**